pkg/proxy/util/utils.go

```diff
@@ -56,3 +65,20 @@ func ShouldSkipService(svcName types.NamespacedName, service *api.Service) bool
 	}
 	return false
 }
+
+// This assumes ip is a valid IP.
+func IsDualStackIP(ip string, isIPv6Mode bool) bool {
```

Comment:
This is an awkward API, to me.  Maybe it's just the name.  What if this were "IsCorrectIPVersion(ip string, expectIPv6 bool)" ?  It's not really about dual-stack at all, but about not matching the expected version.  Would eliminate "dual stack" from all the names

Reply:
Ack, I chose a wrong name. Will remove "dual stack"s.

Reply:
Done.